HIP/hip-1056.md

```diff
@@ -0,0 +1,1742 @@
+a file it expected to be either a raw protobuf `Block` message with file name of `<BLOCK_NUMBER>.blk` e.g.
+`0000000000000000001.blk` . With total of 19 digits to allow for 2^63 blocks. Or a compressed file e.g.
+`0000000000000000001.blk.gz` or `0000000000000000001.blk.ztsd` .
+
+### Important Note Regarding Map Fields
+
+The network will never add a `map` to a protobuf message in the Block Stream, and will not use a `map` for any value in
+state. The protocol buffer implementations across different languages do not, and cannot, guarantee iteration order for
+map entries and some implementations deliberately randomize the iteration order to "aggressively" prevent any code
+from depending on map iteration order.
+
+This unreliable iteration order has the consequence that block validation would fail non-deterministically when
+encountering a different iteration order because the serialized bytes would not be in the same sequence and would not
+generate the same hash. All implementations of block validation would be required to use custom protobuf libraries,
+which would be required to use *the same* deterministic iteration order when serializing bytes, so that the generated
+hash value would match every other implementation. The network values the freedom to implement a compliant Block Stream
+consumer using any valid protocol buffer library and any computing language of choice, and therefore chooses not to use
+the `map` type in protobuf messages.
+
+### Protobuf Definitions
+
+The following protobuf specification defines a `Block` and it’s components. Notably, the protobufs in this HIP have
+partial comments for brevity and where they depend on existing message types defined in HAPI specification they are not
+repeated here. The full and detailed specification is present in the public
+[hedera-protobufs github](https://github.com/hashgraph/hedera-protobufs/tree/main/block) repository.
+
+#### Block
+
+A collection of block items that represent a single block of transactions on the network.
+
+```protobuf
+/**
+ * Block is used to serialize all the data for a block into record stream of
+ * block files. This structure represents a block on Hedera.
+ */
+message Block {
+    /**
+     * The items contained in the block that make up the running hash.
+     */
+    repeated BlockItem items = 1;
+}
+```
+
+The order of block items in a block stream is important in some cases it designates useful positions in the stream that
+a client may make inferences on. Particularly the start of a `Block`, `Event` and `Transaction` are useful
+considerations, in addition to the end of a `Block` or `Transaction`.
+
+#### BlockItem
+
+A block item is a logical separation of components that together make up a block. A Block item may be one of the 10 types
+noted below.
+
+```protobuf
+message BlockItem {
+    oneof item {
+        /**
+         * An header for the block, marking the start of a new block.
+         */
+        BlockHeader block_header = 1;
```

Comment:
BlockHeader and EventHeader missing package so not clear they are input or output

Reply:
This is correct in the `hedera-protobufs` repository, so we probably need to resync this HIP content to that codebase.
Note, there's a [pull request](https://github.com/hashgraph/hedera-protobufs/pull/421) open there to make some additional changes required based on implementation experience (e.g. making the `BlockItems` API use `repeated` instead of singular to improve network efficiency).